core/src/forwarding_stage.rs

```diff
@@ -10,19 +10,33 @@ use {
     solana_poh::poh_recorder::PohRecorder,
     solana_sdk::pubkey::Pubkey,
     solana_streamer::sendmmsg::batch_send,
+    solana_tpu_client_next::transaction_batch::TransactionBatch,
     std::{
         iter::repeat,
         net::{SocketAddr, UdpSocket},
         sync::{Arc, RwLock},
         thread::{Builder, JoinHandle},
     },
+    tokio::sync::mpsc,
 };
 
+#[derive(Clone)]
+pub enum TransactionForwardClient {
+    ConnectionCache(Arc<ConnectionCache>),
+    TpuClientNextSender(mpsc::Sender<TransactionBatch>),
+}
+
+impl From<Arc<ConnectionCache>> for TransactionForwardClient {
+    fn from(cache: Arc<ConnectionCache>) -> Self {
+        TransactionForwardClient::ConnectionCache(cache)
+    }
+}
+
 pub struct ForwardingStage<T: LikeClusterInfo> {
     receiver: BankingPacketReceiver,
     poh_recorder: Arc<RwLock<PohRecorder>>,
     cluster_info: T,
-    connection_cache: Arc<ConnectionCache>,
+    transaction_client: TransactionForwardClient,
     data_budget: DataBudget,
     udp_socket: UdpSocket,
```

Comment:
I think that logic for networking should be split:
* one simple struct for votes (UDP)
* one structure for other transactions (TPU)